### PR TITLE
(Revival) Allows slime extracts to act as upgrades for stuff around the station

### PR DESCRIPTION
**Pull request by ghost**
This refactors into slime_act as requested, which wasn't done leading to the PR's closure (#27569)
## Original text of PR:
[content] [tested]

This adds several uses for xenobio products around the station. This should encourage people to do it more often. I think that's positive.

While all the changes are tested and working, I would appreciate input on if there is a more efficient implementation for any of the changes.

:cl:
* rscadd: Green Slime Extract can be applied to hydroponics trays to prevent aging
* rscadd: Yellow Slime Extract can be applied to an RPD to enable it's autowrench feature. Rightclick on the RPD to toggle
* rscadd: Metal Slime Extract can be applied to an RPD to enable multilayer building. Rightclick on the RPD to toggle
* rscadd: Black Slime Extract can be applied to chemistry dispenser to make creating doctor's delight a little easier
* rscadd: Blue Slime Extract can be applied to foam fire extinguishers to make them really cold
* rscadd: Orange Slime Extract can be applied to a pepper spray to make it refill automatically
* rscadd: Oil Slime Extract can be applied to any pickaxe or drill to make it mine in an AoE
* rscadd: Bluespace Slime Extract can be applied to a socket wrench, allowing it to vent the excess gas into space, instead of into the room

### CODE REVIEW

**Comment by Durgen** [comment]
pink slime extract could be something for clown?

Maybe it could upgrade the bike horn or the rainbow crayon. Or clowns could apply it to eguns to turn them into one of the meme clumsy guns.

**Comment by DeityLink** [comment]
Somehow I feel like plant ageing should be cerulean (the one that increases core uses) rather than green (that turns people into slime). Something about core use increase feels like it fits the concept of plants living longer.

**Comment by Eneocho** [comment]
> 
> Somehow I feel like plant ageing should be cerulean (the one that increases core uses) rather than green (that turns people into slime). Something about core use increase feels like it fits the concept of plants living longer.

Green doesn't really fit, since you can already get slimetox and advanced slimetox with the exotic seeds.

The hydroponics one feels a bit underwhelming when compared to the rest since basic xenobotany gets you plants that can compensate their aging with ridiculous output and growth speed, also clonex on a plant is a thing. Why not make it so it passively raises one random good stat per tick (some good values for this would be: +0.05 yield, +0.1 endurance, +0.5 potency if under 150/+0.1 if over 150, +0.1 lifespan). Alternatively, instead of helping with botany, it could help with apiculture, like making the beehives self harvest.

Also, are the multilayer and autowrench RPD upgrades mutually exclusive or can I have both active/available at the same time?

File: code/modules/reagents/machinery/chem_dispenser.dm [hunk 1]
```diff
@@ -339,6 +337,14 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 			to_chat(user, "You can't add \a [D] to the machine while the panel is open.")
 			return
 
+/obj/machinery/chem_dispenser/slime_act(primarytype, mob/user)
+	..()
+	if(primarytype == /mob/living/carbon/slime/black)
+		has_slime=1
+		dispensable_reagents.Add(DSYRUP)
+		to_chat(user, "You throw the slime into the dispenser's tank.")
```
Comment (Eneocho) [inline code/modules/reagents/machinery/chem_dispenser.dm:345]: "You scan the slime with the dispenser's manipulators, unlocking a new reagent." or along those lines would be better and move obvious.

**Comment by jwhitak** [comment]
~~What happens if you right click an RPD with both mutlilayer and autowrench?~~

Disregard.

**Comment by KillFurries** [comment]
What the slime extracts do seem too specific and not general enough to see use. Who is going to contact a xenobiologist to get their hands on a slime extract to make their wrench vent gas into space?
Orange slime for an automatic pepper spray refill? When was the last time anyone used pepper spray?
It's very rare that anyone breeds slimes and these weird small upgrades that will quickly fade into obscurity aren't worth the effort to get. If you're a doctor who wants to create Doctor's Delight, it takes more effort to get your hands on slime extract than to break into the bar and steal their machines.
More general upgrades like putting the slime on a weapon, a slime for upgrading the station pets or to reinforce doors/walls would encourage people to use the slimes more.

**Comment by gurfan** [comment]
All you have to do is walk to science and ask the xenobiologist there for an extract

File: code/modules/reagents/machinery/chem_dispenser.dm (hunk 1 above)
Comment (ghost) [reply code/modules/reagents/machinery/chem_dispenser.dm:345]: eneocho moment

**Comment by Durgen** [comment]
> What the slime extracts do seem too specific and not general enough to see use. Who is going to contact a xenobiologist to get their hands on a slime extract to make their wrench vent gas into space?

I would contact a xenobio for yellow, metal, blue, and bluespace as engineer or atmos

I bet it would work more like, xenobio emerging from his autism cave after an hour of frantic jobbing to dispense gifts to the crew.

**Comment by Killette3** [comment]
Would the blue extract FFE's be able to one shot hypothermia people?

**Comment by Eneocho** [comment]
> 
> 
> Would the blue extract FFE's be able to one shot hypothermia people?

I'm guessing it'd be equal to being suddenly spaced, since it's around the same temperature.

File: code/modules/RCD/RPD.dm [hunk 2]
```diff
@@ -136,6 +139,103 @@
...
+	if(~selected.flags & RCD_SELF_SANE && get_energy(user) < selected.energy_cost) // Handle energy amounts, but only if not SELF_SANE.
+		return 1
+	if(build_all && istype(selected, /datum/rcd_schematic/pipe))
+		var/datum/rcd_schematic/pipe/our_schematic = selected //typecast
+		if(our_schematic.layer) // this is needed because disposal pipe schematic datums are retarded
+			for(var/layer in 1 to 5)
+				busy  = TRUE // Busy to prevent switching schematic while it's in use.
+				our_schematic.set_layer(layer)
+				if(layer != 5 )
+					spawn(-1) our_schematic.attack(A, user)
```
Comment (Kurfursten) [inline code/modules/RCD/RPD.dm]: no one line spawn, also why -1?

Comment (ghost) [reply code/modules/RCD/RPD.dm]: I fixed that one line, also not sure, didn't code the -1 so don't know the significance.

**Pull request merged**